### PR TITLE
[chore] introduce helm installation test

### PR DESCRIPTION
**Pull request by odubajDT**
## Changes
- added helm installation workflow with install script and example values.yaml file
- cleaned up unused helm installation files
- test is ran only on push into `main` branch, green run [here](https://github.com/Dynatrace/dynatrace-otel-collector/actions/runs/11908419967/job/33183882876?pr=374)

### CODE REVIEW

File: internal/testbed/scripts/install_dt_collector_helm.sh [hunk 1]
```diff
@@ -2,17 +2,25 @@
...
+
+helm upgrade -i --wait dynatrace-collector open-telemetry/opentelemetry-collector \
+    -f tmp.yaml \
     --timeout 5m
 
 # show the collector logs
-kubectl -n otel-collector logs -l app.kubernetes.io/name=opentelemetry-collector
+kubectl logs -l app.kubernetes.io/name=opentelemetry-collector | grep "Everything is ready. Begin running and processing data." || exit 1
+
+rm tmp.yaml
```
Comment (odubajDT) [inline internal/testbed/scripts/install_dt_collector_helm.sh:26]:
```suggestion
rm tmp.yaml

```

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -128,6 +128,21 @@ jobs:
...
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - name: Setup cluster
+        uses: ./.github/actions/create-cluster
+
+      - name: Run helm installation
+        run: |
+          REPOSITORY="dynatrace-otel-collector" TAG="e2e-test" ./internal/testbed/scripts/install_dt_collector_helm.sh || exit 1;
```
Comment (mowies) [inline .github/workflows/e2e.yaml:144]: you could put the 2 env vars into an `env:` block here
```suggestion
          REPOSITORY="dynatrace-otel-collector" TAG="e2e-test" ./internal/testbed/scripts/install_dt_collector_helm.sh || exit 1;
```

File: internal/testbed/scripts/install_dt_collector_helm.sh [hunk 3]
```diff
@@ -2,17 +2,25 @@
...
 
 helm repo add open-telemetry https://open-telemetry.github.io/opentelemetry-helm-charts
 helm repo update
 
-cd .github/actions/deploy-collector
-helm upgrade -i --wait dynatrace open-telemetry/opentelemetry-collector \
-    -f collector-helm-values.yaml \
-    -n otel-collector \
+export TAG=$TAG
+export REPOSITORY=$REPOSITORY
```
Comment (mowies) [inline internal/testbed/scripts/install_dt_collector_helm.sh:11]: i'm not sure if this is necessary for envsubst here honestly...

File: config_examples/collector-helm-values.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  exporters:
+    otlphttp:
+      endpoint: "${env:DT_ENDPOINT}"
+      headers:
+        Authorization: "Api-Token ${env:DT_API_TOKEN}"
+  extensions:
+    health_check:
+      endpoint: "0.0.0.0:13133"
+  service:
+    extensions: [ health_check ]
```
Comment (mowies) [inline config_examples/collector-helm-values.yaml:38]:
```suggestion
    extensions: [health_check]
```

File: internal/testbed/scripts/install_dt_collector_helm.sh (hunk 3 above)
Comment (odubajDT) [reply internal/testbed/scripts/install_dt_collector_helm.sh]: works without it, removed

**Review by mowies — APPROVED** [review]

File: config_examples/collector-helm-values.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        http:
+          endpoint: ${env:MY_POD_IP}:4318
+  exporters:
+    otlphttp:
+      endpoint: "${env:DT_ENDPOINT}"
+      headers:
+        Authorization: "Api-Token ${env:DT_API_TOKEN}"
+  extensions:
+    health_check:
+      endpoint: "0.0.0.0:13133"
```
Comment (bacherfl) [inline config_examples/collector-helm-values.yaml:36]:
```suggestion
      endpoint: "${env:MY_POD_IP}:13133"
```

**Review by bacherfl — APPROVED** [review]

**Pull request merged**